Gems/ROS2Controllers/Code/Source/RobotControl/Controllers/RigidBodyController/RigidBodyTwistControlComponent.cpp

```diff
@@ -72,23 +88,103 @@ namespace ROS2Controllers
             },
             aznumeric_cast<int32_t>(AzPhysics::SceneEvents::PhysicsStartFinishSimulationPriority::Components));
         sceneInterface->RegisterSceneSimulationFinishHandler(defaultSceneHandle, m_sceneFinishSimHandler);
         AZ::TickBus::Handler::BusDisconnect();
     }
 
+    AZ::Vector3 ComputeImpulse(const AZ::Vector3& errorVec, AZStd::array<PidConfiguration, 3>& controllers, float fixedDeltaTime)
+    {
+        AZ::Vector3 impulse = AZ::Vector3::CreateZero();
+        for (size_t i = 0; i < 3; ++i)
+        {
+            const auto error = errorVec.GetElement(i);
+            impulse.SetElement(i, controllers[i].ComputeCommand(error, fixedDeltaTime * 1'000'000'000));
+        }
+        return impulse;
+    }
+
+    void RigidBodyTwistControlComponent::OnSceneSimulationFinish(AzPhysics::SceneHandle sceneHandle, float fixedDeltaTime)
+    {
+        AzPhysics::SceneInterface* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(sceneInterface, "No scene interface");
+
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        AZ_Assert(defaultSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid default physics scene handle");
+
+        if (m_bodyHandle == AzPhysics::InvalidSimulatedBodyHandle)
+        {
+            // This component is only interested in the default scene
+            return;
+        }
+        auto* rigidBody = sceneInterface->GetSimulatedBodyFromHandle(sceneHandle, m_bodyHandle);
+        AZ_Assert(sceneInterface, "No body found for previously given handle");
+
+        const AZ::Transform robotTransform = rigidBody->GetTransform();
+
+        if (m_config.m_physicalApi == RigidBodyTwistControlComponentConfig::PhysicalApi::Kinematic)
+        {
+            // Convert local steering to world frame
+            const auto linearVelocityGlobal = robotTransform.TransformVector(m_linearVelocityLocal);
+            const auto angularVelocityGlobal = robotTransform.TransformVector(m_angularVelocityLocal);
+
+            // Set the kinematic target for the rigid body
+            // This will move the rigid body to the target position and orientation
+            Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequests::SetKinematic, true);
+
+            AZ::Transform kinematicTarget = robotTransform;
+            kinematicTarget.SetTranslation(kinematicTarget.GetTranslation() + linearVelocityGlobal * fixedDeltaTime);
+            kinematicTarget.SetRotation(
+                AZ::Quaternion::CreateFromScaledAxisAngle(angularVelocityGlobal * fixedDeltaTime) * kinematicTarget.GetRotation());
+
+            Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequests::SetKinematicTarget, kinematicTarget);
+            return;
+        }
+        if (m_config.m_physicalApi == RigidBodyTwistControlComponentConfig::PhysicalApi::Velocity)
```

Comment:
```suggestion
        }
        else if (m_config.m_physicalApi == RigidBodyTwistControlComponentConfig::PhysicalApi::Velocity)
```
`else if` improves the readability, as different implementations exclude each other (you might also consider adding additional `else` at the end to trigger an error in case unimplemented API is selected)